common/containers/Tabs/ScheduleTransaction/components/Fields/WindowStart/WindowStartFieldFactory/WindowStartFieldFactory.tsx

```diff
@@ -0,0 +1,72 @@
+import { Query } from 'components/renderCbs';
+import { setCurrentWindowStart, TSetCurrentWindowStart } from 'actions/schedule';
+import { WindowStartInputFactory } from './WindowStartInputFactory';
+import React from 'react';
+import { connect } from 'react-redux';
+import { ICurrentWindowStart } from 'selectors/schedule';
+import { getLatestBlock } from 'selectors/config';
+import { AppState } from 'reducers';
+
+interface DispatchProps {
+  setCurrentWindowStart: TSetCurrentWindowStart;
+}
+
+interface OwnProps {
+  windowStart: string | null;
+  latestBlock: string;
+  withProps(props: CallbackProps): React.ReactElement<any> | null;
+}
+
+export interface CallbackProps {
+  isValid: boolean;
+  readOnly: boolean;
+  currentWindowStart: ICurrentWindowStart;
+  onChange(ev: React.FormEvent<HTMLInputElement>): void;
+}
+
+type Props = DispatchProps & OwnProps;
+
+class WindowStartFieldFactoryClass extends React.Component<Props> {
+  public componentDidMount() {
+    const { latestBlock, windowStart } = this.props;
+
+    if (windowStart) {
+      this.props.setCurrentWindowStart(windowStart);
+    } else {
+      this.props.setCurrentWindowStart(latestBlock);
```

Comment:
We're going to need to kick off a fetch for latest block somewhere. Right now I think it only gets fetched when you change nodes. Looks like the default value is `'???'`.

Reply:
Alternatively we could just start with an empty field.

Reply:
I've removed the default value and we'll start with an empty field.